core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -100,6 +141,20 @@ public long length() {
       return len;
     }
 
+    @Override
+    public long sizeBytes() {
+      if (sizeBytes == null) {
+        this.sizeBytes = FileScanTask.super.sizeBytes();
+      }
+
+      return sizeBytes;
+    }
+
+    @Override
+    public int filesCount() {
+      return fileScanTask.filesCount();
```

Comment:
Delegate to the parent task to avoid materializing the list of deletes for the files count in the split task as well.